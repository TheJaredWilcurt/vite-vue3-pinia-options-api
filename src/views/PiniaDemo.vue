<template>
  <div>
    <h2 class="green">üçç Pinia Example:</h2>

    <ul>
      <li>
        <strong class="pinia-label">Count:</strong>
        <span data-test="count">
          {{ count }}
        </span>
      </li>
      <li>
        <strong class="pinia-label">Doubled Count:</strong>
        <span data-test="doubled">
          {{ doubledCount }}
        </span>
      </li>
      <li>
        <button
          data-test="increment"
          @click="incrementCount"
        >
          Increment Count
        </button>
      </li>
    </ul>
  </div>
</template>

<script>
import { mapActions, mapState } from 'pinia';

import { counterStore } from '@/stores/counter.js';

export default {
  name: 'PiniaDemo',
  methods: {
    ...mapActions(counterStore, [
      'incrementCount'
    ])
  },
  computed: {
    ...mapState(counterStore, [
      'count',
      'doubledCount'
    ])
  }
};
</script>

<style>
.pinia-label {
  margin-right: 0.25em;
}
</style>
