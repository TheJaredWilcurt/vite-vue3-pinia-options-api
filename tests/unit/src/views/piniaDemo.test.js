import { flushPromises } from '@vue/test-utils';
import PiniaDemo from '@/views/PiniaDemo.vue';

import testHelpers from '@@/unit/testHelpers.js';

describe('PiniaDemo', () => {
  const setupWrapper = async () => {
    const options = {};
    const wrapper = await testHelpers.mount(PiniaDemo, options);
    return wrapper;
  };

  test('Mounts and renders', async () => {
    const wrapper = await setupWrapper();

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Counter increments', async () => {
    const wrapper = await setupWrapper();

    expect(wrapper.find('[data-test="count"]').text())
      .toEqual('2');

    expect(wrapper.find('[data-test="doubled"]').text())
      .toEqual('4');

    const button = wrapper.find('[data-test="increment"]');
    console.log(button.html());
    await button.trigger('click');
    await flushPromises();

    expect(wrapper.find('[data-test="count"]').text())
      .toEqual('3');

    expect(wrapper.find('[data-test="doubled"]').text())
      .toEqual('6');
  });
});
