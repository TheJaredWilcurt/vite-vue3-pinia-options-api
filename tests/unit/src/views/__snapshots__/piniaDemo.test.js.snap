// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PiniaDemo > Mounts and renders 1`] = `
<div>
  <h2 class="green">
    🍍 Pinia Example:
  </h2>
  <ul>
    <li>
      <strong class="pinia-label">
        Count:
      </strong>
      <span>
        2
      </span>
    </li>
    <li>
      <strong class="pinia-label">
        Doubled Count:
      </strong>
      <span>
        4
      </span>
    </li>
    <li>
      <button>
        Increment Count
      </button>
    </li>
  </ul>
</div>
`;
